### ISSUE TITLE
[Bug] Spawned patient got to Cast Remover instead of Reception Desk

### ISSUE THREAD

**Issue by ghost** [CorsixTH/CorsixTH#2241]
### Describe the issue

Patient got stuck inside a room, making nurse to stuck as well (sleep counter was still working). Impossible to kick both patient or nurse. As consequence, room couldn't be edited.

### Steps to Reproduce

1. *Load save game attached to this ticket*

2. Check the room.
[12.zip](https://github.com/CorsixTH/CorsixTH/files/9587092/12.zip)
[gamelog.txt](https://github.com/CorsixTH/CorsixTH/files/9587093/gamelog.txt)

### Expected Behaviour

Normal behaviour :P.

### Save Game

Save game attached to report: Yes

### CorsixTH Version

0.66

### Operating System

Windows 10

### Theme Hospital Version

GOG.com

### Gamelog.txt

```bash
Compiled with api_version: 2682, arch: x64, audio: true, os: windows
SDL renderer: direct3d
Lua 5.4 run with api version: 2682, game version: v0.66, savegame version: 170

Savegame version is 170 (v0.66), originally it was 170 (v0.66).
Warning: Trying to remove nonexistent room build callback (function: 0000027E14D5F9E0) from humanoid (table: 0000027E192B9870).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E199B08A0) from humanoid (table: 0000027E192B9870).
Warning: Trying to remove nonexistent staff callback (function: 0000027E14D607C0) from humanoid (table: 0000027E192B9870).
Warning: Trying to resolve door deadlock at (86, 49)

Empty action queue!
Last action: walk
stack traceback:
	Lua\entities\humanoid.lua:630: in method '_handleEmptyActionQueue'
	Lua\entities\humanoid.lua:469: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
	Lua\entity.lua:216: in method 'tick'
	Lua\world.lua:1085: in method 'onTick'
	Lua\app.lua:1130: in function <Lua\app.lua:1127>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Error in timer handler: 
Lua\graphics.lua:337: bad argument #1 to 'sizeOf' (string expected, got table)
stack traceback:
	[C]: in method 'sizeOf'
	Lua\graphics.lua:337: in method 'sizeOf'
	Lua\dialogs\confirm_dialog.lua:60: in local 'constructor'
	Lua\class.lua:74: in global 'UIConfirmDialog'
	Lua\entities\humanoid.lua:632: in method '_handleEmptyActionQueue'
	Lua\entities\humanoid.lua:469: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
	Lua\entity.lua:216: in method 'tick'
	Lua\world.lua:1085: in method 'onTick'
	Lua\app.lua:1130: in function <Lua\app.lua:1127>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Recovering from error in timer handler...

Empty action queue!
Last action: walk
stack traceback:
	Lua\entities\humanoid.lua:630: in method '_handleEmptyActionQueue'
	Lua\entities\humanoid.lua:469: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
	Lua\entity.lua:216: in method 'tick'
	Lua\world.lua:1085: in method 'onTick'
	Lua\app.lua:1130: in function <Lua\app.lua:1127>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Error in timer handler: 
Lua\graphics.lua:337: bad argument #1 to 'sizeOf' (string expected, got table)
stack traceback:
	[C]: in method 'sizeOf'
	Lua\graphics.lua:337: in method 'sizeOf'
	Lua\dialogs\confirm_dialog.lua:60: in local 'constructor'
	Lua\class.lua:74: in global 'UIConfirmDialog'
	Lua\entities\humanoid.lua:632: in method '_handleEmptyActionQueue'
	Lua\entities\humanoid.lua:469: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
	Lua\entity.lua:216: in method 'tick'
	Lua\world.lua:1085: in method 'onTick'
	Lua\app.lua:1130: in function <Lua\app.lua:1127>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Recovering from error in timer handler...
Warning: Trying to remove nonexistent room build callback (function: 0000027E10745250) from humanoid (table: 0000027E192BB5B0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E194AE680) from humanoid (table: 0000027E192BB5B0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E10744E30) from humanoid (table: 0000027E192BB5B0).
Warning: Trying to remove nonexistent room build callback (function: 0000027E14D53C80) from humanoid (table: 0000027E192BA8B0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E0A6DDFC0) from humanoid (table: 0000027E192BA8B0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E14D54CA0) from humanoid (table: 0000027E192BA8B0).
Warning: Trying to remove nonexistent room build callback (function: 0000027E159B7C60) from humanoid (table: 0000027E192A8AB0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E154C2D50) from humanoid (table: 0000027E192A8AB0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E159B83E0) from humanoid (table: 0000027E192A8AB0).
Warning: Trying to remove nonexistent room build callback (function: 0000027E15FAF1A0) from humanoid (table: 0000027E1145ECA0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E19657C50) from humanoid (table: 0000027E1145ECA0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E15FAE420) from humanoid (table: 0000027E1145ECA0).
Warning: Trying to remove nonexistent room build callback (function: 0000027E159B9700) from humanoid (table: 0000027E153AFC30).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E194B3030) from humanoid (table: 0000027E153AFC30).
Warning: Trying to remove nonexistent staff callback (function: 0000027E159B9040) from humanoid (table: 0000027E153AFC30).
Warning: Trying to remove nonexistent room build callback (function: 0000027E159BC1C0) from humanoid (table: 0000027E098A3FA0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E09B54E60) from humanoid (table: 0000027E098A3FA0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E159BBF20) from humanoid (table: 0000027E098A3FA0).
Warning: Trying to remove nonexistent room build callback (function: 0000027E14D608E0) from humanoid (table: 0000027E192B8C30).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E199B05D0) from humanoid (table: 0000027E192B8C30).
Warning: Trying to remove nonexistent staff callback (function: 0000027E14D60D60) from humanoid (table: 0000027E192B8C30).
Warning: Trying to remove nonexistent room build callback (function: 0000027E159BC100) from humanoid (table: 0000027E192B2070).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E10DF0FF0) from humanoid (table: 0000027E192B2070).
Warning: Trying to remove nonexistent staff callback (function: 0000027E159BBFE0) from humanoid (table: 0000027E192B2070).
Warning: Trying to remove nonexistent room build callback (function: 0000027E159CACE0) from humanoid (table: 0000027E098AE2E0).
Warning: Trying to remove nonexistent room remove callback (function: 0000027E15AA6CD0) from humanoid (table: 0000027E098AE2E0).
Warning: Trying to remove nonexistent staff callback (function: 0000027E159CAC20) from humanoid (table: 0000027E098AE2E0).
Error in keydown handler: 
Lua\map.lua:64: bad argument #1 to 'floor' (number expected, got nil)
stack traceback:
	[C]: in function 'math.floor'
	Lua\map.lua:64: in method 'getRoomId'
	Lua\world.lua:2292: in method 'getRoom'
	Lua\room.lua:77: in method 'getEntranceXY'
	Lua\room.lua:95: in method 'createEnterAction'
	Lua\humanoid_actions\seek_room.lua:106: in upvalue 'action_seek_room_goto_room'
	Lua\humanoid_actions\seek_room.lua:287: in local 'callback'
	Lua\entities\humanoid.lua:825: in field 'notifyNewRoom'
	Lua\entities\humanoids\patient.lua:864: in method 'notifyNewRoom'
	Lua\world.lua:824: in method 'markRoomAsBuilt'
	Lua\game_ui.lua:1200: in field 'callback'
	Lua\ui.lua:790: in field 'onKeyDown'
	Lua\game_ui.lua:340: in function <Lua\game_ui.lua:339>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Error in buttonup handler: 
Lua\map.lua:64: bad argument #1 to 'floor' (number expected, got nil)
stack traceback:
	[C]: in function 'math.floor'
	Lua\map.lua:64: in method 'getRoomId'
	Lua\world.lua:2292: in method 'getRoom'
	Lua\room.lua:77: in method 'getEntranceXY'
	Lua\room.lua:95: in method 'createEnterAction'
	Lua\humanoid_actions\seek_room.lua:106: in upvalue 'action_seek_room_goto_room'
	Lua\humanoid_actions\seek_room.lua:287: in local 'callback'
	Lua\entities\humanoid.lua:825: in field 'notifyNewRoom'
	Lua\entities\humanoids\patient.lua:864: in method 'notifyNewRoom'
	Lua\world.lua:824: in method 'markRoomAsBuilt'
	Lua\game_ui.lua:1200: in method 'setEditRoom'
	Lua\game_ui.lua:642: in function <Lua\game_ui.lua:623>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
Error in timer handler: 
Lua\map.lua:64: bad argument #1 to 'floor' (number expected, got nil)
stack traceback:
	[C]: in function 'math.floor'
	Lua\map.lua:64: in method 'getRoomId'
	Lua\world.lua:2292: in method 'getRoom'
	Lua\room.lua:77: in method 'getEntranceXY'
	Lua\room.lua:95: in method 'createEnterAction'
	Lua\humanoid_actions\meander.lua:52: in field '?'
	Lua\entities\humanoid.lua:476: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\answer_call.lua:40: in field '?'
	Lua\entities\humanoid.lua:476: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
	Lua\entity.lua:216: in field 'tick'
	Lua\entities\humanoids\staff.lua:126: in field 'tick'
	Lua\entities\humanoids\staff\doctor.lua:97: in method 'tick'
	Lua\world.lua:1085: in method 'onTick'
	Lua\app.lua:1130: in function <Lua\app.lua:1127>
	(...tail calls...)
	Lua\app.lua:1030: in function <Lua\app.lua:1025>
```

**Comment by mugmuggy** [comment]
I had a quick look at this, the patient appears to have not visited any rooms or anything. They passed reception and I assume is on way to a GPs Office, but they appear to have entered the fracture clinic and that is causing the rest of the issues. Going to be pretty difficult to ascertain what went wrong I think.

**Comment by lewri** [comment]
Apparently this patient had the Sleeping Illness. No idea why you'd want to have a nap in the cast remover.

```
{
  [ [[th]] ]=userdata: 00000218E71F4028,
  [ [[vaccinated]] ]=false,
  [ [[noqueue_ticks]] ]=1,
  [ [[health_history]] ]={
    0.81,
    0.806,
    0.802,
    0.798,
    0.794,
    0.79,
    0.786,
    0.782,
    0.778,
    0.854,
    0.85,
    0.846,
    0.842,
    0.838,
    0.834,
    0.83,
    0.826,
    0.822,
    0.818,
    0.814,
    [ [[last]] ]=9,
    [ [[size]] ]=20
  },
  [ [[die_anims]] ]={
    [ [[hands_east]] ]=2446,
    [ [[rise_hell_east]] ]=384,
    [ [[fly_east]] ]=2450,
    [ [[wings_east]] ]=2438,
    [ [[rise_east]] ]=2434,
    [ [[fall_east]] ]=1682
  },
  [ [[should_knock_on_doors]] ]=true,
  [ [[ticks]] ]=false,
  [ [[falling_anim]] ]=1682,
  [ [[cure_rooms_visited]] ]=0,
  [ [[active_moods]] ]={
    [ [[exit]] ]={...}
  },
  [ [[insurance_company]] ]=2,
  [ [[world]] ]={
 ... -- removed this
  },
  [ [[tap_foot_anim]] ]=2960,
  [ [[shake_fist_anim]] ]=392,
  [ [[staff_change_callbacks]] ]={
  },
  [ [[needs_redirecting]] ]=false,
  [ [[attributes]] ]={
    [ [[health]] ]=0.774,
    [ [[happiness]] ]=0.76100063079342,
    [ [[thirst]] ]=0.30452896727346,
    [ [[toilet_need]] ]=0.30984044339997,
    [ [[warmth]] ]=0.27491571726602
  },
  [ [[hospital]] ]={
   ... -- removed this
  },
  [ [[speed]] ]=[[normal]],
  [ [[diagnosis_progress]] ]=0,
  [ [[reserved_for]] ]=false,
  [ [[diagnosed]] ]=false,
  [ [[dynamic_info]] ]={
    [ [[text]] ]={...}
  },
  [ [[available_diagnosis_rooms]] ]={
    [[general_diag]],
    [[cardiogram]],
    [[scanner]],
    [[ultrascan]],
    [[blood_machine]],
    [[x_ray]],
    [[psych]],
    [[ward]]
  },
  [ [[previous_action]] ]={
    [ [[walking_to_vaccinate]] ]=false,
    [ [[truncate_only_on_high_priority]] ]=false,
    [ [[path_x]] ]={...},
    [ [[is_entering]] ]=false,
    [ [[is_leaving]] ]=false,
    [ [[x]] ]=68,
    [ [[saved_must_happen]] ]=false,
    [ [[must_happen]] ]=true,
    [ [[path_index]] ]=2,
    [ [[name]] ]=[[walk]],
    [ [[on_interrupt]] ]=function: 00000218BF0C98A0,
    [ [[on_restart]] ]=function: 00000218BF1176D0,
    [ [[path_y]] ]={...},
    [ [[no_truncate]] ]=false,
    [ [[y]] ]=92
  },
  [ [[humanoid_class]] ]=[[Standard Male Patient]],
  [ [[has_fallen]] ]=1,
  [ [[build_callbacks]] ]={
  },
  [ [[vomit_anim]] ]=2056,
  [ [[mood_info]] ]=userdata: 00000218E71F6998,
  [ [[door_anims]] ]={
    [ [[knock_east]] ]=288,
    [ [[knock_north]] ]=286,
    [ [[leaving]] ]=182,
    [ [[leaving_swing]] ]=2040,
    [ [[entering]] ]=184,
    [ [[entering_swing]] ]=2042
  },
  [ [[going_home]] ]=true,
  [ [[get_up_anim]] ]=384,
  [ [[treatment_history]] ]={
  },
  [ [[infected]] ]=false,
  [ [[in_room]] ]={
    [ [[room_info]] ]={...},
    [ [[door]] ]={...},
    [ [[objects]] ]={...},
    [ [[sound_played]] ]=true,
    [ [[world]] ]={...},
    [ [[built]] ]=true,
    [ [[is_active]] ]=true,
    [ [[hospital]] ]={...},
    [ [[humanoids]] ]={...},
    [ [[y]] ]=90,
    [ [[x]] ]=67,
    [ [[width]] ]=4,
    [ [[humanoids_enroute]] ]={...},
    [ [[crashed]] ]=false,
    [ [[height]] ]=4,
    [ [[id]] ]=51,
    [ [[maximum_patients]] ]=1
  },
  [ [[remove_callbacks]] ]={
  },
  [ [[animation_idx]] ]=18,
  [ [[dead]] ]=false,
  [ [[hover_cursor]] ]={
    [ [[draw]] ]=function: 00000218E732A8E0
  },
  [ [[action_queue]] ]={
    {...},
    {...},
    {...},
    {...},
    {...},
    {...}
  },
  [ [[pee_anim]] ]=2244,
  [ [[playing_sounds_in_random_sequence]] ]=false,
  [ [[tile_x]] ]=68,
  [ [[tile_y]] ]=92,
  [ [[going_to_toilet]] ]=[[no]],
  [ [[last_room]] ]=<reference loop>,
  [ [[waiting_for_sound_effects_to_be_turned_on]] ]=false,
  [ [[walk_anims]] ]={
    [ [[walk_north]] ]=16,
    [ [[walk_east]] ]=18,
    [ [[idle_north]] ]=24,
    [ [[idle_east]] ]=26
  },
  [ [[action_string]] ]=<LocalisedString> Current value:Sent Home,
  [ [[disease]] ]={
    [ [[yawn]] ]=true,
    [ [[cure]] ]=<LocalisedString> Current value:Cure - A high dosage of powerful stimulant is administered by a Nurse.,
    [ [[emergency_number]] ]=18,
    [ [[symptoms]] ]=<LocalisedString> Current value:Symptoms - Overwhelming desire to crash out everywhere.,
    [ [[cause]] ]=<LocalisedString> Current value:Cause - overactive sleep gland in the roof of the mouth.,
    [ [[visuals_id]] ]=11,
    [ [[diagnosis_rooms]] ]={...},
    [ [[initPatient]] ]=function: 00000218BF1C1AC0,
    [ [[contagious]] ]=true,
    [ [[name]] ]=<LocalisedString> Current value:Sleeping Illness,
    [ [[expertise_id]] ]=13,
    [ [[emergency_sound]] ]=[[emerg015.wav]],
    [ [[cure_price]] ]=750,
    [ [[id]] ]=[[sleeping_illness]],
    [ [[treatment_rooms]] ]={...}
  },
  [ [[attempted_to_infect]] ]=false,
  [ [[last_move_direction]] ]=[[south]],
  [ [[litter_countdown]] ]=4,
  [ [[on_ground_anim]] ]=1258,
  [ [[has_vomitted]] ]=1,
  [ [[check_watch_anim]] ]=2964,
  [ [[cured]] ]=false,
  [ [[layers]] ]={
    2,
    2,
    0,
    0,
    [ 0 ]=6
  },
  [ [[has_passed_reception]] ]=true,
  [ [[yawn_anim]] ]=368,
  [ [[animation_flags]] ]=1,
  [ [[vaccination_candidate]] ]=false
}
```

**Comment by Xcl4m4t10n** [comment]
He wanted to sleep so badly that he fake an illness knowing that he will be sit on that machine.

https://www.youtube.com/watch?v=mJtoQpU6O3g

Nice discovery!

**Comment by lewri** [comment]
Can we investigate this one any further?

**Comment by lewri** [comment]
Nothing to proceed on here really without a before problem save. Users account deleted so we can't get that.

**Comment by ARGAMX** [comment]
Looks very similar to #2501
There is Cast Remover on the path to the reception desk. So spawned patient is getting into this room.
Fixed by #2762

<img width="330" height="232" alt="Image" src="https://github.com/user-attachments/assets/298889e8-e611-4cfd-b9e4-7f5663305acb" />
